core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -278,36 +281,40 @@ class TransactionStateManager(brokerId: Int,
     val topicPartition = new TopicPartition(Topic.TRANSACTION_STATE_TOPIC_NAME, partitionId)
 
     inLock(stateLock) {
+      leavingPartitions.remove(partitionId)
```

Comment:
Edit: thinking a bit more on lock protection during loading, I think it is simpler to just drop the best-effort optimization to access the non-thread safe `loadingPartitions` as all other access are going to be protected by the read-write lock, and hence making the set as synchronized set just for this own case seem to be a waste.